gitpod-io/gitpod
# [usage] add optional UserID to filter usage

## Description


Filters usage data for members to only their own usage.

| Owner | Member |
| ------- | -------- |
| <img width="1371" alt="Screenshot 2023-08-07 at 16 37 52" src="https://github.com/gitpod-io/gitpod/assets/372735/9d82f0cd-dc8e-4ffb-80b1-46e91e39293a"> | <img width="1140" alt="Screenshot 2023-08-07 at 16 37 44" src="https://github.com/gitpod-io/gitpod/assets/372735/44a17da4-28c7-492c-a200-cddbac540f3e"> |

## Related Issue(s)


Fixes EXP-357

## How to test


- Join this org ([invite link](https://se-usage-for-user.preview.gitpod-dev.com/orgs/join?inviteId=b597fe58-77f0-400e-b944-040a919ee30d)) (which already has usage)
- check no usage in the usage view
- start and stop a workspace
- wait for reconciliation and verify the usage record is there.

## Documentation


#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - se-usage-for-user</li>
	<li><b>🔗 URL</b> - <a href="https://se-usage-for-user.preview.gitpod-dev.com/workspaces" target="_blank">se-usage-for-user.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - se-usage-for-user-gha.14830</li>
</ul>

## Build Options

<details>
<summary>Build</summary>

      Run the build with werft instead of GHA
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

</details>

<details>
<summary>Installer</summary>

  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

      If enabled this will build `install/preview`
      If enabled this will create the environment on GCE infra
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
</details>

/hold
